NPE in JAXRS client when OpenTracing is included

18.0.0.2

FROM open-liberty:webProfile7

I have a simple service that acts as a JAXRS client to another service.

When I include the microprofile-1.3 feature, I get an NPE after a warning about OpenTracing. (see attached trace)

Everything works fine with microprofile-1.2 and it also works if I use only the constituent features of microprofile-1.3 that the app needs (mpFaultTolerance-1.0,mpHealth-1.0,mpConfig-1.2,mpRestClient-1.0,mpMetrics-1.1,mpJwt-1.0,mpOpenAPI-1.0)

The code is here: https://github.com/eclipse/microprofile-service-mesh-service-a

[trace.log](https://github.com/OpenLiberty/open-liberty/files/2197643/trace.log)

@fmhwong I think the bug is here ... https://github.com/OpenLiberty/open-liberty/blob/6367193a1b75a4b70543255ea4e5625ca18e6229/dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingClientFilter.java#L158

I think a null check is needed for when the property just doesn't exist in the `clientRequestContext`.

```
[18/07/18 11:36:41:741 BST] 000000d2 SystemErr                                                    R Caused by: java.lang.NullPointerException
[18/07/18 11:36:41:741 BST] 000000d2 SystemErr                                                    R 	at com.ibm.ws.opentracing.OpentracingClientFilter.filter(OpentracingClientFilter.java:158)
[18/07/18 11:36:41:741 BST] 000000d2 SystemErr                                                    R 	at org.apache.cxf.jaxrs.client.spec.ClientResponseFilterInterceptor.handleMessage(ClientResponseFilterInterceptor.java:66)
[18/07/18 11:36:41:741 BST] 000000d2 SystemErr                                                    R 	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
[18/07/18 11:36:41:741 BST] 000000d2 SystemErr                                                    R 	at org.apache.cxf.jaxrs.client.ClientMessageObserver.onMessage(ClientMessageObserver.java:56)
```

--- a/dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingClientFilter.java
+++ b/dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingClientFilter.java
@@ -155,7 +155,8 @@ public class OpentracingClientFilter implements ClientRequestFilter, ClientRespo
                        ClientResponseContext clientResponseContext) throws IOException {
         String methodName = "filter(incoming)";
 
-        if ((Boolean) clientRequestContext.getProperty(CLIENT_SPAN_SKIPPED_ID)) {
+        Boolean skip = (Boolean) clientRequestContext.getProperty(CLIENT_SPAN_SKIPPED_ID);
+        if ((skip != null) && skip) {
             if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
                 Tr.debug(tc, methodName, "skipped");
             }

